staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -236,12 +236,6 @@ func PatchResource(r rest.Patcher, scope *RequestScope, admit admission.Interfac
 		}
 		trace.Step("Object stored in database")
 
-		if err := setObjectSelfLink(ctx, result, req, scope.Namer); err != nil {
```

Comment:
stopping calculating the selflink field is fine, but does anything ensure selfLink is set to "" in the returned object (for example, clearing any selfLink value set by the patch)?

Reply:
Set object selflink was previously (before this PR):
(a) setting selflinks
(b) ensuring that `Item` field of the list is non-empty
Neither of them is needed for patch operation now (I claim that it wasn't needed before too, because the code setting selflinks was also code from a generic code, so it wasn't needed from individual handlers (and it wasn't there for create or update - just for patch).

Regarding setting selfLink to empty - etcd code is always clearing it before writing in etcd, so we should be fine.

Reply:
> I claim that it wasn't needed before too, because the code setting selflinks was also code from a generic code, so it wasn't needed from individual handlers (and it wasn't there for create or update - just for patch

create would call FillObjectMetaSystemFields, which cleared self-link. if update wasn't doing anything to clear selfLink, that does sound inconsistent

> etcd code is always clearing it before writing in etcd, so we should be fine

I agree for persisted data. I'm less sure about data sent to admission webhooks, which happens pre-persistence and pre-PrepareObjectForStorage.

Reply:
> create would call FillObjectMetaSystemFields, which cleared self-link. if update wasn't doing anything to clear selfLink, that does sound inconsistent

Yes - it was inconsistent. 
But setSelflink didn't clean selflink too - so this PR isn't changing anything (this is happening after persistence, to after PrepareObjectForStorage so wasn't needed anyway).

> I agree for persisted data. I'm less sure about data sent to admission webhooks, which happens pre-persistence and pre-PrepareObjectForStorage.

But this place of code is happening after persistence anyway...

Reply:
ok